sync-core/src/main/java/com/cloudant/sync/query/IndexManager.java

```diff
@@ -0,0 +1,261 @@
+                }
+            }
+        } catch (SQLException e) {
+            logger.log(Level.SEVERE, "Failed to get a list of indexes in the database.", e);
+        } finally {
+            DatabaseUtils.closeCursorQuietly(cursor);
+        }
+
+        return indexes;
+    }
+
+    /**
+     *  Add a single, possibly compound, index for the given field names.
+     *
+     *  This method generates a name for the new index.
+     *
+     *  @param fieldNames List of field names in the sort format
+     *  @return name of created index
+     */
+    public String ensureIndexed(ArrayList<Object> fieldNames) {
+        logger.log(Level.SEVERE, "ensureIndexed(fieldNames) not implemented.");
+
+        return null;
+    }
+
+    /**
+     *  Add a single, possibly compound, index for the given field names.
+     *
+     *  This function generates a name for the new index.
+     *
+     *  @param fieldNames List of field names in the sort format
+     *  @param indexName Name of index to create
+     *  @return name of created index
+     */
+    public String ensureIndexed(ArrayList<Object> fieldNames, String indexName) {
+        return ensureIndexed(fieldNames, indexName, "json");
+    }
+
+    /**
+     *  Add a single, possibly compound, index for the given field names.
+     *
+     *  This function generates a name for the new index.
+     *
+     *  @param fieldNames List of field names in the sort format
+     *  @param indexName Name of index to create
+     *  @param indexType "json" is the only supported type for now
+     *  @return name of created index
+     */
+    public String ensureIndexed(ArrayList<Object> fieldNames, String indexName, String indexType) {
+        if (fieldNames == null || fieldNames.isEmpty()) {
+            return null;
+        }
+
+        if (indexName == null || indexName.isEmpty()) {
+            return null;
+        }
+
+        if (indexType == null || !indexType.equalsIgnoreCase("json")) {
+            return null;
+        }
+
+        return IndexCreator.ensureIndexed(fieldNames,
+                                          indexName,
+                                          indexType,
+                                          database,
+                                          datastore,
+                                          queue);
+    }
+
+    /**
+     *  Delete an index.
+     *
+     *  @param indexName Name of index to delete
+     *  @return deletion status as true/false
+     */
+    public boolean deleteIndexNamed(String indexName) {
+        boolean success = true;
+
+        // TODO - implement method
+        // Drop the index table
+        // Delete the metadata entries
+
+        logger.log(Level.SEVERE, "deleteIndexNamed(indexName) not implemented.");
+
+        return false;
+    }
+
+    /**
+     *  Update all indexes.
+     *
+     *  @return update status as true/false
+     */
+    public boolean updateAllIndexes() {
+
+        // TODO
```

Comment:
This comment was a straight rip from CDTQIndexManager updateAllIndexes.  I tried to keep the comments as consistent between the two versions as possible.  @mikerhodes, is this a placeholder TODO for additional work we plan on later or can it be removed?